src/main/java/uk/nhs/digital/nhsconnect/nhais/translator/amendment/mappers/AmendmentDateOfBirthToEdifactMapper.java

```diff
@@ -0,0 +1,52 @@
+package uk.nhs.digital.nhsconnect.nhais.translator.amendment.mappers;
+
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.nhs.digital.nhsconnect.nhais.exceptions.PatchValidationException;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.PersonDateOfBirth;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.Segment;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentBody;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentPatch;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentPatchOperation;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentValue;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.JsonPatches;
+import uk.nhs.digital.nhsconnect.nhais.service.TimestampService;
+
+@Component
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class AmendmentDateOfBirthToEdifactMapper extends AmendmentToEdifactMapper {
+
+    private final TimestampService timestampService;
+
+    @Override
+    Segment mapPatches(AmendmentBody amendmentBody) {
+        var dateOfBirth = amendmentBody.getJsonPatches().getBirthDate()
+            .map(AmendmentPatch::getValue)
+            .map(AmendmentValue::get)
+            .map(timestampService::parseDate)
+            .orElseThrow(() -> new PatchValidationException("Missing " + JsonPatches.BIRTH_DATE_PATH));
+
+        return PersonDateOfBirth.builder()
+            .timestamp(dateOfBirth)
+            .build();
+    }
+
+    @Override
+    boolean shouldCreateSegment(AmendmentBody amendmentBody) {
+        return amendmentBody.getJsonPatches().getBirthDate().isPresent();
+    }
+
+    @Override
+    void validatePatches(JsonPatches patches) {
+        if (patches.getBirthDate().isPresent()) {
```

Comment:
Functional way:
````java
if (patches.getBirthDate()
            .map(AmendmentPatch::isRemoval)
            .isPresent()) {	
	throw new PatchValidationException("Illegal remove operation on " + JsonPatches.BIRTH_DATE_PATH));
}
if (patches.getBirthDate()
            .map(AmendmentPatch::getValue)
            .map(AmendmentValue::get)
            .filter(String::isBlank)
            .isPresent()) {
            throw new PatchValidationException("Invalid value for " + JsonPatches.BIRTH_DATE_PATH);
}
````
or use `ifPresent(value -> {throw ...});`

Reply:
No, I tried that already. If value is null then it doesn't do the validation as `map` exits in case of null

Reply:
You're right, then at least extract `patches.getBirthDate().get()` to variable